### PR TITLE
DAOS-11831 test: Add restart after bringing up network interface

### PR DESCRIPTION
**Pull request by shimizukko**
After bringing up the network interface, some ranks may become excluded. If that happens, restart the servers and check again. If they're still not joined, fail the test.

Skip-unit-tests: true
Skip-fault-injection-test: true
Test-tag: target_failure_with_rp network_failure_isolation Test-repeat: 5
Required-githooks: true

Signed-off-by: Makito Kano <makito.kano@intel.com>

### Before requesting gatekeeper:

* [ ] Two review approvals and any prior change requests have been resolved.
* [ ] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [ ] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [ ] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [x] You are the appropriate gatekeeper to be landing the patch.
* [x] The PR has 2 reviews by people familiar with the code, including appropriate watchers.
* [x] Githooks were used. If not, request that user install them and check copyright dates.
* [x] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [x] All builds have passed.  Check non-required builds for any new compiler warnings.
* [x] Sufficent testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [x] If applicable, the PR has addressed any potential version compatibility issues.
* [x] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [x] Extra checks if forced landing is requested
  * [x] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [x] Quick-build or Quick-functional is not used.
* [x] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

### CODE REVIEW

**Review by dinghwah — DISMISSED** [review]
lgtm

**Review by daltonbohning — DISMISSED** [review]

File: src/tests/ftest/deployment/network_failure.py [hunk 1]
```diff
@@ -448,11 +448,20 @@ def test_network_failure_isolation(self):
...
             time.sleep(20)
 
-        # Some ranks may be excluded after bringing up the network interface, so wait
-        # until they are up (joined).
-        if not self.wait_for_ranks_to_join():
-            msg = "One or more servers crashed after bringing up the network interface!"
-            errors.append(msg)
+        # Some ranks may be excluded after bringing up the network interface. Check if
+        # all ranks are joined. If not, restart the servers and check again.
+        time.sleep(60)
```
Comment (phender) [inline src/tests/ftest/deployment/network_failure.py:453]: We do not want to use large sleeps in our test code.  Instead, we should query at a much smaller interval with an overall timeout.   This way we don't waste cycles when the test could be progressing.  Something similar to https://github.com/daos-stack/daos/blob/master/src/tests/ftest/util/command_utils.py#L711-L726 except we issue `check_system_query_status(dmg_command.system_query())` in the while loop with a `sleep(1)` if not all ranks are joined.

**Review by phender — CHANGES_REQUESTED** [review]

**Review by phender — APPROVED** [review]

File: src/tests/ftest/deployment/network_failure.py (hunk 1 above)
Comment (shimizukko) [reply src/tests/ftest/deployment/network_failure.py]: Done. Thanks.

**Review by daltonbohning — APPROVED** [review]

**Review by dinghwah — APPROVED** [review]
lgtm

**Review by phender — APPROVED** [review]

**Pull request merged**